File: backend/recycler-service/src/main/java/com/ecobridge/recycler/service/RecyclerService.java
package com.ecobridge.recycler.service;

import com.ecobridge.recycler.dto.request.CreateRecyclerRequest;
import com.ecobridge.recycler.dto.request.UpdateRecyclerRequest;
import com.ecobridge.recycler.dto.response.RecyclerResponse;

import java.util.List;
import java.util.UUID;

public interface RecyclerService {

    RecyclerResponse createRecycler(CreateRecyclerRequest request);

    RecyclerResponse getRecyclerById(UUID id);

    List<RecyclerResponse> getMyRecycler();

    RecyclerResponse updateRecycler(UUID id, UpdateRecyclerRequest request);

    void deleteRecycler(UUID id);

    List<RecyclerResponse> getAllRecyclers();

    List<RecyclerResponse> getEligibleRecyclers();

    RecyclerResponse reduceCapacity(
            UUID recyclerId,
            Double quantity
    );
}

File: frontend/ecobridge-frontend/src/services/recyclerService.js
import recyclerApi from "../api/recyclerApi";

export const createRecycler = (data) => {
    return recyclerApi.post("/api/v1/recyclers", data);
};

export const getMyRecycler = () => {
    return recyclerApi.get("/api/v1/recyclers/my");
};

export const getRecyclerById = (id) => {
    return recyclerApi.get(`/api/v1/recyclers/${id}`);
};

export const updateRecycler = (id, data) => {
    return recyclerApi.put(`/api/v1/recyclers/${id}`, data);
};